src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  let tmp = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      tmp = {
+        ...tmp, ...action.extraData,
+      };
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete tmp[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      tmp = {};
+    }
```

Comment:
Good work but let's rewrite it to switch/case statement since we rely on single action.type value